### DISCUSSION TITLE
Call spawnSync shell command from within seeds.ts?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [prisma/prisma#16529] [Q&A] [answered]
I have 2 prisma clients and try to do something like this

```
const organisations = await Promise.all(
  Array.from({ length: 3 }, async (value, index) => {
    let objectId = String((index + 1));
    console.log("seed organisation " + objectId);
    const organisation = await prismaPublic.organisation.create({data: {name: "Org #" + objectId}
    });

    await provisionPgSchema([organisation.id](http://organisation.id/));

    // const user = await prismaWorkspace.user.create({```

Is this the right approach? 

await provisionPgSchema method does this:

```
import { spawnSync } from "node:child_process";

export function provisionTenant(tenantId: string) {
  console.log("Provisioning organisation", tenantId);
  spawnSync("prisma/genPgSchema", ["organisation", organisationId]);
  console.log("Provisioning complete");
}

```

This hangs the excution.

Can I approach it this way or is this totally wrong?

To me this sounded like logic since I need a pg schema creation from the first seeds.ts file before it can continue to the seeds.ts of the 2nd client ( it seeds the created pg schema ).

**Comment by nurul3101** [comment, marked as answer]
Hello @Jarrodsz 👋 

It seems like you want to run a shell command programmatically, can you have a look at this [comment](https://github.com/prisma/prisma/issues/4703#issuecomment-669801976) on this GitHub Issue: #4703. It describes running a command through `spawn` which might help in solving this issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
